extensions/interactions/ItemSelectionInput/ItemSelectionInput.py

```diff
@@ -94,3 +98,239 @@ class ItemSelectionInput(base.BaseInteraction):
+                cls._convert_does_not_contains_at_least_one_of_rule_spec_to_proto), # pylint: disable=line-too-long
+        }
+
+        rule_type_to_proto_mapping = {
+            'Equals': lambda x: (
+                state_pb2.ItemSelectionInputInstanceDto.RuleSpecDto(equals=x)),
+            'ContainsAtLeastOneOf': lambda x: (
+                state_pb2.ItemSelectionInputInstanceDto.RuleSpecDto(
+                    contains_at_least_one_of=x)),
+            'DoesNotContainAtLeastOneOf': lambda x: (
+                state_pb2.ItemSelectionInputInstanceDto.RuleSpecDto(
+                    does_not_contain_at_least_one_of=x)),
+            'IsProperSubsetOf': lambda x: (
+                state_pb2.ItemSelectionInputInstanceDto.RuleSpecDto(
+                    is_proper_subset_of=x))
+        }
+
+        for rule_spec in rule_specs_list:
+            rule_type = rule_spec.rule_type
+            rule_proto = (
+                rule_type_to_proto_func_mapping[rule_type](
+                    rule_spec.inputs['x']
+                )
+            )
+            rule_specs_list_proto.append(
+                rule_type_to_proto_mapping[rule_type](rule_proto)
+            )
+
+        return rule_specs_list_proto
+
+    @classmethod
+    def _convert_equals_rule_spec_to_proto(cls, choice_list):
+        """Creates a proto object for EqualsSpecDto.
+
+        Args:
+            choice_list: list. Choice list to select from.
+
+        Returns:
+            EqualsSpecDto. The proto object.
+        """
+        return RULE_SPEC_DTO.EqualsSpecDto(
+            input=cls._convert_set_of_translatable_html_content_ids_to_proto(
+                choice_list)
+        )
+
+    @classmethod
+    def _convert_contains_at_least_one_of_rule_spec_to_proto(cls, choice_list):
+        """Creates a proto object for ContainsAtLeastOneOfSpecDto.
+
+        Args:
+            choice_list: list. Choice list to select from.
+
+        Returns:
+            ContainsAtLeastOneOfSpecDto. The proto object.
+        """
+        return RULE_SPEC_DTO.ContainsAtLeastOneOfSpecDto(
+            input=cls._convert_set_of_translatable_html_content_ids_to_proto(
+                choice_list)
+        )
+
+    @classmethod
+    def _convert_is_proper_subset_of_rule_spec_to_proto(cls, choice_list):
+        """Creates a proto object for IsProperSubsetOfSpecDto.
+
+        Args:
+            choice_list: list. Choice list to select from.
+
+        Returns:
+            IsProperSubsetOfSpecDto. The proto object.
+        """
+        return RULE_SPEC_DTO.IsProperSubsetOfSpecDto(
+            input=cls._convert_set_of_translatable_html_content_ids_to_proto(
+                choice_list)
+        )
+
+    @classmethod
+    def _convert_does_not_contains_at_least_one_of_rule_spec_to_proto(
+        cls, choice_list
+    ):
+        """Creates a proto object for DoesNotContainAtLeastOneOfSpecDto.
+
+        Args:
+            choice_list: list. Choice list to select from.
```

Comment:
Here and elsewhere: when the type is list, we usually write list(...) (where the "..." should be filled in). Please do this throughout.

Reply:
Fixed all the lists with their types.